src/trace/trace.jl

```diff
@@ -94,14 +81,10 @@ function forkc(trace :: Trace)
   newtrace = typeof(trace)()
   newtrace.task = Base.copy(trace.task)
   newtrace.spl = trace.spl
-  if trace.spl != nothing
-    gid = trace.spl.alg.group_id
-  else
-    gid = 0
-  end
 
-  n_rand = min(trace.vi.index, length(groupvals(trace.vi, gid, trace.spl)))
-  newtrace.vi = retain(deepcopy(trace.vi), gid, n_rand, trace.spl)
+  n_rand = min(trace.vi.index, length(groupvals(trace.vi, trace.spl)))
+  newtrace.vi = deepcopy(trace.vi)
+  retain!(newtrace.vi, n_rand, trace.spl)
```

Comment:
We cannot use `end` here because we don't have `ndims(VarInfo)`. However, we can't even implement this `ndims` because the total length is dependent on the sampler. So we have to write things like

```julia
l = length(getranges(newtrace.vi, spl))
newtrace.vi[spl, n_rand+1:l] = NaN
```

This extra line of code doesn't looks neat.